src/tools/diagnostics.md

```diff
@@ -0,0 +1,2436 @@
+---
+title: Diagnostics
+description: Details for diagnostics produced by the Dart analyzer.
+---
+
+This page lists all diagnostic messages produced by the Dart analyzer,
+with details about what those messages mean
+and how you can fix your code.
+For more information about the analyzer, see
+[Customizing static analysis](/guides/language/analysis-options).
+
+
+## Glossary
+
+This page uses the following terms.
+
+### Element
+
+[PENDING: definition]
+
+### Potentially non-nullable
+
+A type is _potentially non-nullable_ if it's either explicitly non-nullable or
+if it's a type parameter.
+The latter case is included because
+the actual runtime type might be non-nullable.
+
+
+## Compile-time errors
+
+The analyzer produces the following diagnostics when [PENDING...].
+
+### AMBIGUOUS_SET_OR_MAP_LITERAL_BOTH
+
+#### Message
+
+This literal contains both 'Map' and 'Iterable' spreads, which makes it impossible to determine whether the literal is a map or a set.
+
+#### Description
+
+Because map and set literals use the same delimiters (`{` and `}`), the analyzer looks at the type arguments and the elements to determine which kind of literal you meant. When there are no type arguments and all of the elements are spread elements (which are allowed in both kinds of literals), then the analyzer uses the types of the expressions that are being spread. If all of the expressions have the type `Iterable`, then it's a set literal; if they all have the type `Map`, then it's a map literal.
+
+The analyzer produces this diagnostic when some of the expressions being spread have the type `Iterable` and others have the type `Map`, making it impossible for the analyzer to determine whether you are writing a map literal or a set literal.
+
+#### Example
+
+The following code produces this diagnostic:
+
+{% prettify dart %}
+union(Map<String, String> a, List<String> b, Map<String, String> c) => [!{...a, ...b, ...c}!];
+{% endprettify %}
```

Comment:
Is there a reason we weren't using the \`\`\` dart for syntax highlighting this codeblock? We're losing syntax highlighting only for this codeblock and it's a little confusing to read:

<img width="714" alt="Screen Shot 2019-05-31 at 3 49 04 PM" src="https://user-images.githubusercontent.com/5550175/58738680-af056480-83bb-11e9-9a11-fc1e8158fd53.png">

Reply:
IIRC, highlights didn't work if I used the ```dart syntax. We should fix this.